src/ui/EditorNS/editor.cpp

```diff
@@ -206,10 +236,27 @@ namespace EditorNS
 
     QList<QMap<QString, QString>> Editor::languages()
     {
+        // TODO: Cache this data.
+        QJSEngine engine;
+        QFileInfo fileInfo(Notepadqq::editorPath());
+        QString fileName = fileInfo.absolutePath() + "/classes/Languages.js";
```

Comment:
This method actually increases startup time by a margin... waiting for the entire web browser to be loaded before getting the languages doesn't make sense (Values that never change shouldn't require UI to be loaded to be retrieved).

It just seemed like a cleaner way to do it... I'm not against changing it, but in essence this method isn't crippled by requiring a widget to be initialized.